pkg/kubelet/status/manager.go

```diff
@@ -139,14 +139,27 @@ func (m *manager) Start() {
 func (m *manager) GetPodStatus(uid types.UID) (api.PodStatus, bool) {
 	m.podStatusesLock.RLock()
 	defer m.podStatusesLock.RUnlock()
-	status, ok := m.podStatuses[m.podManager.TranslatePodUID(uid)]
-	return status.status, ok
+	cachedStatus, ok := m.podStatuses[m.podManager.TranslatePodUID(uid)]
+	if !ok {
+		return api.PodStatus{}, false
+	}
+	// Return a copy.
```

Comment:
Won't copying in `SetPodStatus` before writing enough? I don't think we need to return a new copy on read.